bindings/python/test/test_mongocrypt.py

```diff
@@ -64,15 +66,46 @@ def test_mongocrypt_binary_out(self):
 
 class TestMongoCryptOptions(unittest.TestCase):
 
-    def test_mongocrypt_options(self):
+    def test_mongocrypt_options_nonlocal_kms(self):
         schema_map = bson_data('schema-map.json')
         valid = [
             ({'aws': {'accessKeyId': '', 'secretAccessKey': ''}}, schema_map),
-            ({'aws': {'accessKeyId': 'foo', 'secretAccessKey': 'foo'}}, None),
+            ({'aws': {'accessKeyId': 'foo', 'secretAccessKey': 'foo'}}, None)]
+        for kms_providers, schema_map in valid:
+            opts = MongoCryptOptions(kms_providers, schema_map)
+            self.assertEqual(opts.kms_providers, kms_providers)
+            self.assertEqual(opts.schema_map, schema_map)
+
+    @unittest.skipIf(PY3, 'python 2 only')
+    def test_mongocrypt_options_local_kms_py2(self):
+        valid = [
             ({'local': {'key': b'1'*96}}, None),
             ({'aws': {'accessKeyId': 'foo', 'secretAccessKey': 'foo'},
               'local': {'key': b'1'*96}}, None),
+            ({'local': {'key': unicode_type(base64.b64encode('1'*96))}}, None)
         ]
+        for kms_providers, schema_map in valid:
+            opts = MongoCryptOptions(kms_providers, schema_map)
+            for provider_name in kms_providers:
+                map = kms_providers[provider_name]
+                if provider_name == 'local':
+                    if isinstance(map['key'], unicode_type):
+                        self.assertEqual(
+                            opts.kms_providers[provider_name], map)
+                    else:
+                        expected = {'key': base64.b64encode(map['key'])}
+                        self.assertEqual(
+                            opts.kms_providers[provider_name], expected)
+                else:
+                    self.assertEqual(opts.kms_providers[provider_name], map)
+            self.assertEqual(opts.schema_map, schema_map)
+
+    @unittest.skipIf(not PY3, 'python 3 only')
+    def test_mongocrypt_options_local_kms_py3(self):
+        valid = [
+            ({'local': {'key': b'1'*96}}, None),
+            ({'aws': {'accessKeyId': 'foo', 'secretAccessKey': 'foo'},
+              'local': {'key': b'1'*96}}, None)]
```

Comment:
Shouldn't we test base64 encoding on Python 3 too?

Reply:
Good point. Done.